Support returning Result type from async fn main

Seems like this is broken in 1.1.0:

```rust
#[smol_potat::main]
async fn main() -> anyhow::Result<()> {
```

```
error[E0308]: mismatched types
   --> src/main.rs:129:20
    |
128 | #[smol_potat::main]
    | -------------------
    | |                 |
    | |                 help: consider removing this semicolon
    | implicitly returns `()` as its body has no tail or `return` expression
129 | async fn main() -> anyhow::Result<()> {
    |                    ^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found `()`
    |
    = note:   expected enum `std::result::Result<(), anyhow::Error>`
            found unit type `()`
```

There doesn't seem to be any tests for this?

--- a/smol-potat-macro/src/lib.rs
+++ b/smol-potat-macro/src/lib.rs
@@ -124,7 +124,7 @@ pub fn main(attr: TokenStream, item: TokenStream) -> TokenStream {
 
             #set_threads
 
-            #crate_root::async_io::block_on(main());
+            #crate_root::async_io::block_on(main())
         }
     };
 

--- a/tests/test.rs
+++ b/tests/test.rs
@@ -1,3 +1,9 @@
+#[smol_potat::main]
+async fn main() -> std::io::Result<()> {
+    assert_eq!(2 * 2, 4);
+    Ok(())
+}
+
 #[smol_potat::test]
 async fn test() -> std::io::Result<()> {
     assert_eq!(2 * 2, 4);

